File: 1662-minimum-numbers-of-function-calls-to-make-target-array/minimum-numbers-of-function-calls-to-make-target-array.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int addOps = 0;   
        int mulOps = 0;   
        for (int x : nums) {
            int curAdd = 0;
            int curMul = 0;
            while (x > 0) {
                if (x % 2 == 1) {   
                    x--;
                    curAdd++;
                } else {            
                    x /= 2;
                    curMul++;
                }
            }

            addOps += curAdd;           
            mulOps = max(mulOps, curMul); 
        }

        return addOps + mulOps;
    }
};

File: 1662-minimum-numbers-of-function-calls-to-make-target-array/minimum-numbers-of-function-calls-to-make-target-array.py
class Solution:
    def minOperations(self, nums):
        add_ops = 0   
        mul_ops = 0   

        for x in nums:
            cur_add = 0
            cur_mul = 0

            while x > 0:
                if x % 2 == 1:
                    x -= 1
                    cur_add += 1
                else:          
                    x //= 2
                    cur_mul += 1

            add_ops += cur_add            
            mul_ops = max(mul_ops, cur_mul) 

        return add_ops + mul_ops
